src/author/author.service.ts

```diff
@@ -1,42 +1,65 @@
-import { Injectable, NotFoundException } from '@nestjs/common';
+import {
+  Injectable,
+  InternalServerErrorException,
+  NotFoundException,
+} from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
 import { Author } from './author.entity';
 import { AuthorDto } from './dto/author.dto';
-import { AuthorsRepository } from './author.repository';
+import { AuthorRepository } from './author.repository';
 import { GetAuthorsFilterDto } from './dto/get-authors-filter.dto';
 
 @Injectable()
 export class AuthorService {
   constructor(
-    @InjectRepository(AuthorsRepository)
-    private authorsRepository: AuthorsRepository,
+    @InjectRepository(AuthorRepository)
+    private authorRepository: AuthorRepository,
   ) {}
 
-  getAuthors(filterDto: GetAuthorsFilterDto): Promise<Author[]> {
-    return this.authorsRepository.getAuthors(filterDto);
+  async getAuthors(filterDto: GetAuthorsFilterDto): Promise<Author[]> {
+    const { search } = filterDto;
+    const query = this.authorRepository.createQueryBuilder('author');
+    if (search) {
+      query.where(`LOWER(Author.name) LIKE LOWER(:search)`, {
+        search: `%${search}%`,
+      });
+    }
+
+    try {
+      const authors = query.getMany();
+      return authors;
+    } catch (error) {
+      throw new InternalServerErrorException();
+    }
   }
 
   async getAuthorById(id: string): Promise<Author> {
-    const found = await this.authorsRepository.findOne({ where: { id } });
+    const found = await this.authorRepository.findOne({ where: { id } });
     if (!found) {
       throw new NotFoundException(`Author with ID ${id} not found`);
     }
     return found;
   }
 
-  createAuthor(authorDto: AuthorDto): Promise<Author> {
-    return this.authorsRepository.createAuthor(authorDto);
+  async createAuthor(authorDto: AuthorDto): Promise<Author> {
+    const { name } = authorDto;
+    const author = this.authorRepository.create({
+      name,
+    });
+    return await this.authorRepository.save(author);
+    // return author;
   }
 
   async updateAuthor(id: string, authorDto: AuthorDto): Promise<Author> {
-    const author = await this.getAuthorById(id);
-    author.name = authorDto.name;
-    await this.authorsRepository.save(author);
-    return author;
+    const author = await this.authorRepository.findOne({ where: { id } });
+    if (!author) {
```

Comment:
const author = await this.authorRepository.findOne({id});

Reply:
done